src/megaclient.cpp

```diff
@@ -12971,6 +12972,177 @@ node_vector *MegaClient::nodesbyfingerprint(FileFingerprint* fingerprint)
     return nodes;
 }
 
+static bool nodes_ctime_less(const Node* a, const Node* b)
+{
+    // heaps return the largest element
+    return a->ctime < b->ctime;
+}
+
+static bool nodes_ctime_greater(const Node* a, const Node* b)
+{
+    return a->ctime > b->ctime;
+}
+
+node_vector MegaClient::getRecentNodes(unsigned maxcount, m_time_t since, bool includerubbishbin)
+{
+    node_vector  v;
+    v.reserve(nodes.size());
+    for (node_map::iterator i = nodes.begin(); i != nodes.end(); ++i)
+    {
+        if (i->second->type == FILENODE && i->second->ctime >= since)
+        {
+            v.push_back(i->second);
+        }
+    }
+
+    // heaps use a 'less' function, and pop_heap returns the largest item stored.
+    std::make_heap(v.begin(), v.end(), nodes_ctime_less);
+
+    node_vector v2;
+    v2.reserve(maxcount);
+    while (v2.size() < maxcount && !v.empty())
+    {
+        std::pop_heap(v.begin(), v.end(), nodes_ctime_less);
+        Node* n = v.back();
+        v.pop_back();
+        if (includerubbishbin || n->firstancestor()->type != RUBBISHNODE)
+        {
+            v2.push_back(n);
+        }
+    }
+    return v2;
+}
+
+
+namespace action_bucket_compare
+{
+    MUTEX_CLASS media_check(false);   // when we can use c++11, switch to a lambda that remembers the MegaClient* for compare().  In the meantime, force just one MegaClient at a time instead.
+    MegaClient* mc;
+
+    const static string webclient_is_image_def = ".jpg.jpeg.gif.bmp.png.";
+    const static string webclient_is_image_raw = ".3fr.arw.cr2.crw.ciff.cs1.dcr.dng.erf.iiq.k25.kdc.mef.mos.mrw.nef.nrw.orf.pef.raf.raw.rw2.rwl.sr2.srf.srw.x3f.";
+    const static string webclient_is_image_thumb = "psd.svg.tif.tiff.webp";  // leaving out .pdf
+    const static string webclient_mime_photo_extensions = ".3ds.bmp.btif.cgm.cmx.djv.djvu.dwg.dxf.fbs.fh.fh4.fh5.fh7.fhc.fpx.fst.g3.gif.heic.heif.ico.ief.jpe.jpeg.jpg.ktx.mdi.mmr.npx.pbm.pct.pcx.pgm.pic.png.pnm.ppm.psd.ras.rgb.rlc.sgi.sid.svg.svgz.tga.tif.tiff.uvg.uvi.uvvg.uvvi.wbmp.wdp.webp.xbm.xif.xpm.xwd.";
+    const static string webclient_mime_video_extensions = ".3g2.3gp.asf.asx.avi.dvb.f4v.fli.flv.fvt.h261.h263.h264.jpgm.jpgv.jpm.m1v.m2v.m4u.m4v.mj2.mjp2.mk3d.mks.mkv.mng.mov.movie.mp4.mp4v.mpe.mpeg.mpg.mpg4.mxu.ogv.pyv.qt.smv.uvh.uvm.uvp.uvs.uvu.uvv.uvvh.uvvm.uvvp.uvvs.uvvu.uvvv.viv.vob.webm.wm.wmv.wmx.wvx.";
```

Comment:
All the above seems to be the supported formats from Webclient. I don't know whether the SDK supports the same set than web, it should, but I think it's not the case.
Should we use the formats supported by the graphics processor / media-info from the SDK? like the ones here:
https://github.com/meganz/sdk/blob/26c7448f57d2b8d6c85a4c4a985ed814f56b0e2d/src/mediafileattribute.cpp#L561
or here:
https://github.com/meganz/sdk/blob/82080a5c43bc5a1bb6a093f0cbbad2d632aef782/src/gfx/freeimage.cpp#L383

Reply:
Alright, just read:
> // evaluate according to the webclient rules, so that we get exactly the same bucketing.

Disregard my previous comment :)

Reply:
Yes, I started out using the SDK facillities, but the requirement is to match the webclient (within reason of course), and there were too many discrepancies reported.   Also there may be native clients that don't want to build with those libraries themselves but still be able to show recent account changes.